plugins/woocommerce/includes/admin/class-wc-admin-log-table-list.php

```diff
@@ -392,4 +487,115 @@ protected function prepare_column_headers() {
 			$this->get_sortable_columns(),
 		);
 	}
+
+	/**
+	 * Convert an array or object into a multidimensional array.
+	 *
+	 * @param array|object $input Array or object to be processed.
+	 *
+	 * @return array
+	 */
+	public function to_multidimensional_array( $input ) {
+		$out   = array();
+		$input = is_object( $input ) ? (array) $input : $input;
+
+		if ( is_array( $input ) ) {
+
+			// Run function on each item in the array.
+			$out = array_map(
+				function( $value ) {
+
+					if ( is_array( $value ) || is_object( $value ) ) {
+						return $this->to_multidimensional_array( $value );
+					}
+
+					return $value;
+				},
+				$input
+			);
+		}
+
+		return $out;
+	}
+
+	/**
+	 * Convert an array to a string with tabbed indentaion.
+	 *
+	 * @param array|object $input   An object or array.
+	 * @param array        $exclude An array of strings of params/keys to exclude.
+	 * @param int          $depth   The current depth of the walker.
+	 *
+	 * @return string
+	 */
+	public function printable_array( $input, $exclude = array(), $depth = 0 ) {
+		$out   = '';
+		$input = is_object( $input ) ? (array) $input : $input;
+		if ( is_array( $input ) ) {
+
+			foreach ( $input as $key => $val ) {
+
+				$key = strip_tags( (string) $key );
+
+				if ( in_array( (string) $key, $exclude ) ) {
+					continue;
+				}
+
+				$tabs = '';
+				for ( $i = 0; $i < $depth; $i++ ) {
+					$tabs .= '  ';
+				}
+
+				$out .= $tabs . '[' . $key . '] => ';
+
+				if ( is_array( $val ) || is_object( $val ) ) {
+					$new_depth = $depth + 1;
+					$out .= '[&#010;' . $this->printable_array( $val, $exclude, $new_depth ) . $tabs . ']';
+				} else {
+					$out .= "'" . $val . "'";
+				}
+				$out .= '&#013;&#010;';
```

Comment:
Might not this also work, and arguably be more readable?

```suggestion
				$out .= "\r\n";
```

Reply:
This works great!  I believe I was originally using single quotes, which just ended up getting printed as plain text.  I thought it was because I was printing it in a text area.